documentation/processing.tex

```diff
@@ -1 +1,315 @@
+
+\subsubsection{Opencode Provider}
+
+Opencode provider is active as long as there are statements in the source file. It retrieves statements from the source code with help of lexer and parser (see \cref{lab06:parser}).
+
+\subsection{Manager's initial state}
+\label{lab06:lib_data}
+Having described processors and providers, we are ready to talk about which of them are initialized by the manager at the beginning of the processing. The manager determines this from library data passed by analyzer.
+
+Library data contain a file name and enumeration telling the meaning of the file that is being parsed --- \emph{processing kind}.
+
+\emph{Ordinary} processing kind states that the file being processed is the main source file (in HLASM called open-code). It is the first file to be processed. With this information, manager initializes all statement providers in respective order and \emph{only} opencode processor. This enumeration is passed when analyzer has owner semantics.
+
+\emph{Copy} and \emph{Macro} processing kinds state that manager will process source code that contains copy or macro definition respectively. Hence, \emph{only} copy definition processor or macro definition processor is initialized. Also, all but macro statement provider are initialized. That is because macro definitions would pollute statement sources (i.e. when parse library provider is called within macro call). Also, no macros will be visited nor needed as a statement source when processing new source code. This enumeration is passed when analyzer has reference semantics.
+
+
+\subsection{Statement field parser}
+\label{lab06:field_parser}
+
+Statement field parser is an interface passed to the statement providers by processing manager implemented by the parser (see \cref{lab06:parser}).
+
+Firstly, it is used during statement passing. In some cases provider is requested a concrete format of a string-stored statement. The string is re-parsed with the according format. Then the field is returned back to the statement provider. 
+
+Another use of field parser is in opencode processor as model statements are resolved there. After variable symbol substitution, the resulting string field is re-parsed with field parser.
+
+\subsection{Instruction processors}
+\label{lab06:instr_proc}
+
+Used by ordinary processor, instruction processors are helper classes that divide processing of HLASM instruction types.
+
+As format of some instruction kinds can be rather complicated, instruction processors contain \emph{Instruction format validators} (see \cref{checker}). They check the statement to validate correctness of used operand format.
+
+During the work of the processors, they encounter \emph{expression} structures (see \cref{ca_expr:logic}, \cref{mach_expr}). Expressions need to be evaluated for processing to be correctly finished.
```

Comment:
co su to tie expression structures? encounter myslis akoze sa s nimi stretnu v zdrojaku?

Reply:
akoze v tom zdrojaku co analyzer parsuje

Reply:
expression structures myslim expressny, nechcel som ich nazvat len expressions tak som dal expression structures.

encounter myslim, tak ako tam je napisane, ze sa stretnu pocas ich prace, teda processingu instrukcii

Reply:
mozno som to len ja ale ked sa povie expression structures tak mi to evokuje C++ reprezentaciu tych expresien. Potom mi to s tym encounter nedava zmysel: ten processor dopredu nevie ze bude pouzivat class mach_expression?